src/com/collibra/command/handlers/AddEdgeCommandHandler.java

```diff
@@ -3,32 +3,39 @@
 import com.collibra.exceptions.NodeNotFoundException;
 import com.collibra.graph.Graph;
 import com.collibra.graph.Node;
-import com.collibra.message.util.ParsedMessage;
 
 import java.io.PrintWriter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import static com.collibra.message.util.MessageUtil.sendMessage;
+import static com.collibra.message.util.CommunicationUtil.sendMessage;
 
 /**
  * Add edge command handler implementation.
  */
 public class AddEdgeCommandHandler implements CommandHandler {
+    private static final Pattern addEdgePattern = Pattern.compile("^ADD EDGE (?<fromNodeName>[A-Za-z0-9\\-]+) " +
+            "(?<toNodeName>[A-Za-z0-9\\-]+) (?<weight>[0-9]+)$");
     private final Graph graph;
 
     public AddEdgeCommandHandler(Graph graph) {
         this.graph = graph;
     }
 
     @Override
-    public void handleCommand(PrintWriter outData, ParsedMessage parsedMessage) {
-        try {
-            graph.addEdge(new Node(parsedMessage.getSourceNodeName()),
-                    new Node(parsedMessage.getDestinationNodeName()),
-                    parsedMessage.getWeight());
-            sendMessage(outData, "EDGE ADDED");
-        } catch (NodeNotFoundException ex) {
-            System.out.println("ERROR: NODE NOT FOUND");
-            sendMessage(outData, "ERROR: NODE NOT FOUND");
+    public void handleCommand(PrintWriter outData, String receivedMessage) {
+        Matcher addEdgeMatcher = addEdgePattern.matcher(receivedMessage);
+        if (addEdgeMatcher.find()) {
+            try {
+                graph.addEdge(new Node(addEdgeMatcher.group("fromNodeName")),
+                        new Node(addEdgeMatcher.group("toNodeName")),
+                        Integer.parseInt(addEdgeMatcher.group("weight")));
+                sendMessage(outData, "EDGE ADDED");
+            } catch (
+                    NodeNotFoundException ex) {
+                System.out.println("ERROR: NODE NOT FOUND");
```

Comment:
The log message should be more descriptive.

Reply:
Done